lib/diaspora_federation/federation/fetcher.rb

```diff
@@ -7,15 +7,9 @@ module Fetcher
       # @param [Symbol, String] entity_type snake_case version of the entity class
       # @param [String] guid guid of the entity to fetch
       def self.fetch_public(author, entity_type, guid)
-        url = DiasporaFederation.callbacks.trigger(
-          :fetch_person_url_to, author, "/fetch/#{entity_name(entity_type)}/#{guid}"
-        )
-        response = HttpClient.get(url)
-        raise "Failed to fetch #{url}: #{response.status}" unless response.success?
-
-        magic_env_xml = Nokogiri::XML(response.body).root
-        magic_env = Salmon::MagicEnvelope.unenvelop(magic_env_xml)
-        Receiver::Public.new(magic_env).receive
+        type = entity_name(entity_type).to_s
+        raise "Already fetching ..." if fetching[type].include?(guid)
+        fetch_from_url(author, type, guid)
       rescue => e # rubocop:disable Lint/RescueWithoutErrorClass
         raise NotFetchable, "Failed to fetch #{entity_type}:#{guid} from #{author}: #{e.class}: #{e.message}"
```

Comment:
Nit: for me it would feel more accurate, if we created a `AlreadyFetching` exception class and propagated it up instead of raising `NotFetchable` for this specific case. What do you think, @SuperTux88 ?

Reply:
No, I do this intentionally, because the only thing `fetch_public` can raise is a `NotFetchable` error. So the only place we need to catch everything when something fails with fetching is here, and everything can just handle `NotFetchable`. The details are logged.

I do the same thing with `DiscoveryError` for when something with the discovery fails.

Yes, I could extend `AlreadyFetching` from `NotFetchable`, so it could be catched with `NotFetchable`. but I don't see what the advantage is when you have a separate class for this case?

Reply:
Well, I mostly disliked the working, since it is not `NotFetchable` when we're already fetching it. But since this exception is public API, then I guess we shouldn't change it for such minor reasons.

Reply:
BTW perhaps this method description needs `@raise NotFetchable` in the description. But I already merged the PR, so whatever :)

Reply:
I added that now directly to develop :)